spec/requests/api/v1/articles_spec.rb

```diff
@@ -13,8 +14,98 @@
     allow(FeatureFlag).to receive(:enabled?).with(:api_v1).and_return(true)
   end
 
-  describe "GET /api/articles" do
-    before { article }
+  describe "GET /articles" do
```

Comment:
> I decided to use a describe block specifically for the API documentation instead of sprinkling it throughout the existing API method docs.

This makes sense

> That brings up a larger question: should these documentation "tests" be in separate classes?

But I think this question is right on point too!

Given that this spec is already ~1200 LOC (yes - it's the largest spec of the API but still) I think it would feel more organized to have all of our `rswag` specs in a separate folder or files. AFAICT folder names are arbitrary up to a point, so I think we could use something we don't currently have, i.e. `spec/rswag/v1/articles_controller_spec.rb`.

That way all our API specs live within `spec/requests/api/v1/*_spec.rb` while all of rswag specs live in `spec/rswag/v1/*_spec.rb`. I haven't tried this before (completely arbitrary folder name), but I think it would work given that we're not using Rspec actual tests (`type: :request/:system/etc`). 

Another idea I just came up with (also not 100% it will work but sort of confident it will 😅 ), is we can have a "twin file" right next to each request spec, but instead of `articles_controller_spec.rb` they could be named `articles_controller_rswag.rb`. Rspec won't execute these files by default (they don't have the `_spec.rb` suffix) and we can run rswag with:

```bash
SWAGGER_DRY_RUN=0 RAILS_ENV=test rails rswag PATTERN="spec/requests/api/v1/**/*_rswag.rb"
```

What are your thoughts on these possibilities?

Reply:
> I think it would feel more organized to have all of our rswag specs in a separate folder or files. AFAICT folder names are arbitrary up to a point, so I think we could use something we don't currently have, i.e. spec/rswag/v1/articles_controller_spec.rb.

Yeah, the rspec folder names are pretty arbitrary (up to a point, there are some rspec looks in, like requests tests in `/request`). Putting the rswag tests in a separate tree under something like `/spec/api-docs` would could be great!

> we can have a "twin file" right next to each request spec

I'm a little less excited by this idea because it means Rspec won't execute these files during a normal test run. There's still some value in them running regularly (especially for older API methods), so I'd like Rspec to catch them normally as well.

Given that these kind of are requests specs, how does placing them in `spec/requests/api/v1/docs` sound? (I could call it `rswag`, but I was reluctant to call out the library name like that)

Reply:
> Given that these kind of are requests specs, how does placing them in spec/requests/api/v1/docs sound? (I could call it rswag, but I was reluctant to call out the library name like that)

`spec/requests/api/v1/docs` sounds great to me! 💯